[XPU]bind elementwise_mod_op_xpu

runzhech opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
New features
### PR changes
OPs
### Describe
Add elementwise_mod OP for XPU along with its UTs

tianshuo78520a committed:
Update Mac cmake version >=3.15 (#41456)

* Update Mac cmake version >=3.15

* notest;read test1

notest;read test2

notest;read test3

* fix inference link error

* fix inference link error

* fix windows link error

* fix cmake_policy

* fix build big size

chenwhql committed:
Add paddle::variant and replace paddle::any (#42139)

* add variant and replace any

* split attribute

pangyoki committed:
disable unittest failed in eager CI in temporary (#42101)

* test=py3-eager

* test=py3-eager

* test=py3-eager

seemingwang committed:
combine graph_table and feature_table in graph_engine (#42134)

* extract sub-graph

* graph-engine merging

* fix

* fix

* fix heter-ps config

* test performance

* test performance

* test performance

* test

* test

* update bfs

* change cmake

* test

* test gpu speed

* gpu_graph_engine optimization

* add dsm sample method

* add graph_neighbor_sample_v2

* Add graph_neighbor_sample_v2

* fix for loop

* add cpu sample interface

* fix kernel judgement

* add ssd layer to graph_engine

* fix allocation

* fix syntax error

* fix syntax error

* fix pscore class

* fix

* change index settings

* recover test

* recover test

* fix spelling

* recover

* fix

* move cudamemcpy after cuda stream sync

* fix linking problem

* remove comment

* add cpu test

* test

* add cpu test

* change comment

* combine feature table and graph table

* test

* test

* pybind

* test

* test

* test

* test

* pybind

* pybind

* fix cmake

* pybind

* fix

* fix

* add pybind

* add pybind

Co-authored-by: DesmonDay <908660116@qq.com>

ronny1996 committed:
[CustomDevice] add eager mode support (#42034)

runzhech committed:
bind elementwise_mod_op_xpu *test=kunlun

runzhech committed:
Merge branch 'PaddlePaddle:develop' into develop

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=42175) <br/>All committers have signed the CLA.

baoachun committed:
fix FlattenContiguousRangeOpConverter out dim error (#42087)

* fix FlattenContiguousRangeOpConverter out dim error

* update code

zhwesky2010 committed:
fix python3.10 compile bug on windows (#42140)

taixiurong commented on paddle/fluid/operators/elementwise/elementwise_mod_op_xpu.cc:
```diff
@@ -0,0 +1,47 @@
+
+REGISTER_OP_XPU_KERNEL(elementwise_mod, ops::ElementwiseModXPUKernel<float>,
+                       ops::ElementwiseModXPUKernel<paddle::platform::float16>);
+
```
api里面还支持了 int32, int64， 补齐一下

taixiurong commented on paddle/fluid/operators/elementwise/elementwise_mod_op_xpu.cc:
```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+
```
日期修改下

taixiurong commented on python/paddle/fluid/tests/unittests/xpu/test_elementwise_mod_op_xpu.py:
```diff
@@ -0,0 +1,96 @@
+            }
+
+            self.attrs = {'axis': 1}
+            self.outputs = {'Out': self.inputs['X'] % self.inputs['Y']}
```
单测规模有点少，可以参考下test_elementwise_mod_op.py 增加一些测试规模

taixiurong commented:
1. PR types 上面的信息描述写一下，补充完整

taixiurong commented:
2. 提交的git 本地的名字和邮箱填写下，不要用root提交，license/cla是过不了的

runzhech committed:
Merge branch 'PaddlePaddle:develop' into develop

chenwhql committed:
Optimize dygraph GetExpectedKernelType perf (#42154)

* opt dygraph scheduling

* revert part impl

tiancaishaonvjituizi committed:
fix incorrect usages of std::move and other compile errors (#41045)

* fix bug of std::move and others

* fix an compile error in debug mode

* fix wrong copy assignment operator

Signed-off-by: tiancaishaonvjituizi <452565578@qq.com>

* reformat

Signed-off-by: tiancaishaonvjituizi <452565578@qq.com>

* reformat

Signed-off-by: tiancaishaonvjituizi <452565578@qq.com>

* fix ArrayRef constructor following llvm

* fix format

* fix conflict with master

chenwhql committed:
fix variant compile error (#42203)

veyron95 committed:
[Eager] Support numpy.ndarry in CastNumpy2Scalar (#42136)

veyron95 committed:
[Eager] Remove redundancy code, fix fp16 case (#42169)

veyron95 committed:
[Eager] Support div(scalar) in eager mode (#42148)

* [Eager] Support div scalar in eager mode

* Updated and remove debug logs

* Remove list, use 'or' directly

* Remove useless statement

sljlp committed:
fix recompute (#42128)

* fix recompute

* modify return

pangyoki committed:
add LICENSE in wheel dist-info package (#42187)

chenwhql committed:
replace any by variant in infermeta (#42181)

BrilliantYuKaimin committed:
【PaddlePaddle Hackathon 2】24、为 Paddle 新增 nn.ChannelShuffle 组网 API (#40743)

* Add infermeta for ChannelShuffle

* Create channel_shuffle_grad_kernel.h

* Create channel_shuffle_kernel.h

* Create channel_shuffle_sig.cc

* Create channel_shuffle_op.cc

ChannelShuffle算子的描述

* Create channel_shuffle_kernel_impl.h

ChannelShuffle核函数的实现

* Create channel_shuffle_grad_kernel_impl.h

ChannelShuffle反向核函数的实现

* Add kernel register of channel shuffle and grad

注册ChannelShuffle及其反向的核函数

* add nn.functional.channel_shuffle

* add nn.ChannelShuffle

* Create test_channel_shuffle.py

* Update example of ChannelShuffle in vision.py

* Update test_channel_shuffle.py

* 修改channel_shuffle核函数的实现位置

* 修正代码格式

* 删除多余空格

* 完善channel_shuffle的错误检查

* Update unary.cc

* Update channel_shuffle_op.cc

* Update test_channel_shuffle.py

* Update unary.cc

* add channel_shuffle

* Update test_channel_shuffle.py

* Update vision.py

* 调整代码格式

* Update channel_shuffle_sig.cc

* 更新ChannelShuffle的文档

* 更新channel_shuffle的文档

* remove ChannelShuffleOpArgumentMapping

* add ChannelShuffleGradInferMeta

* Update channel_shuffle_op.cc

* 调整channel_shuffle及其梯度的核函数的位置

From00 committed:
Do not reset default stream for StreamSafeCUDAAllocator (#42149)

runzhech committed:
add more supported dtypes and UTs *test=kunlun

Commit:
remove redundant computation in Categorical.probs (#42114)

Silv3S committed:
Downloading data for test_analyzer_vit_ocr (#42041)

* Change server URL

* update config

* add test to parallel UT rule

* add checksum to ensure files are downloaded

* change downloading target

* reuse existing variable

* change target directory

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

Yilingyelu committed:
fix en docs of some Apis (gradients, scope_guard, cuda_places, name_scope, device_guard, load_program_state, scale, ParamAttr and WeightNormParamAttr) (#41604)

* Update scope_guard; test=document_fix

* gradients; test=document_fix

* gradients; test=document_fix

* name_scope; test=document_fix

* cpu_places; test=document_fix

* WeightNormParamAttr; test=document_fix

* cuda_places; test=document_fix

* load_program_state; test=document_fix

* device_guard; test=document_fix

* device_guard; test=document_fix

* ParamAttr; test=document_fix

* scale; test=document_fix

* scale; test=document_fix

* update code example；test=document_fix

Co-authored-by: Chen Long <1300851984@qq.com>

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

runzhech committed:
Merge branch 'PaddlePaddle:develop' into develop

runzhech committed:
fix datatype error

runzhech committed:
add op to in xpu1_op_list

runzhech committed:
fix datatype error

add op to in xpu1_op_list

*test=kunlun

add op to in xpu1_op_list

fix elementwise_mod op path error  *test=kunlun

fix elementwise_mod UT error  *test=kunlun

runzhech committed:
fix datatype error

add op to in xpu1_op_list

*test=kunlun

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop *test=kunlun

runzhech committed:
fix elementwise_mod op path error  *test=kunlun

runzhech committed:
fix elementwise_mod UT error  *test=kunlun

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

runzhech committed:
reset PR template file *test=kunlun

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

runzhech committed:
reset PR template

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

runzhech committed:
Merge branch 'PaddlePaddle:develop' into develop

runzhech committed:
Merge branch 'PaddlePaddle:develop' into develop

runzhech committed:
Merge branch 'develop' of https://github.com/runzhech/Paddle into develop

taixiurong reviewed (commented):
LGTM
